### ISSUE TITLE
Issues about issues

### ISSUE THREAD

**Issue by theo-armour** [opentecture/mindmapping#51]
### Story

@afomi 

See #31

> Issues drive Pull Requests. Visible communication happens as a byproduct. ++ the previous text

I very much hear you.

What you describe is a process analogous to the operation of a well-oiled machine. In such operations all parties know where they belong and what they are meant to do. I very much hope we get to such a level before too long.

But we are not there yet. We have few players. And, as far as this player is concerned, he knows not yet where he belongs or what are the good  things to do.

My questions include: how can we in a free, open source manner do the following?

* Create at least a few highly open-ended discussions. Not looking for closing issues. But, metaphorically, looking for openings
* Invent ways of informing newbs that there there is an active, dynamic, friendly community that welcomes new players
* And instill a drive to innovate, break down barriers and help peeps live better

What do we do with issues that may take a long time to close or even that we may never want to close  but keep redefining? And how can do all this using GitHub or something similar?

My own explorations into this include using GitHub pages as blog posts, mucking around with Gists and more. None of these have the email follow through, markdown support and more that GitHub issues provides. So hacking GitHub issues seems natural to me, but I understand it a may well be odd behavior for others.

So I am very open to discussing ways of discussing very open ended discussions using open source tools. ;-)

**Comment by afomi** [comment]
Tools designed for open ended and exploratory discussion like Discourse or
Slack or Discord come to mind.  Sometimes the bits are free, sometimes not.
Usually hosting is required.

The closest thing that comes to mind is to continue to use GitHub issues,
but open the dialog there.  By the time code is written, usually a
discussion is moot. Coordination should happen before code is written and
merged.

Engaging collaborators can happen via many channels.  Understanding their
incentives, origins, and experience is key. And something beyond GitHub.
Dave McClure's Pirate Metrics is a reference to a user funnel worth
consideration.

Also, I'm not sure I see a use case for issues that don't close.

Long term items can live in the roadmap. Active issues can be issues.
 Juggling too much is akin to too much data in RAM. Multitasking is a myth.

I've also seen Google docs or similar for outlining long term product ideas.

Overall, I'm conciously trying to increase focus and limit scope.
Expansive thinking is effortless, but creates implicit obligations I'd
prefer to avoid. I think about the lifecycle from idea to implementation.
 Some ideas must die on the vine. Not all ideas are worth tending to, given
limited resources.

Happy to keep this dialogue open.

**Comment by theo-armour** [comment]
## Communications

* Discourse ~ great but you gotta pay
* Slack and Discord ~ chatting is nice but really is not the same as composing an email in response ti a detailed query
* Google Docs ~ love the collaboration but the dialog not googleable!!

Much more could be said about the above, but GitHub - coders writing for other coders is quite special. If you are a coder.

## Pirate Metrics
http://500hats.typepad.com/500blogs/2007/09/startup-metrics.html

## Focus and scope

Let's focus and limit scope. You specify. I build. Then you build. Repeat until finished.

What code does opentecture need to get the the next stage?

**Comment by afomi** [comment]
Hi @theo-armour, 

The bits of Discourse can be free. Hosting then costs money.

Sounds like you want to talk in Forums or a Mailing List.
GitHub Issues does leverage email, so I see how you're connecting the experience of dialogue and email. Also, you make references to the habits of Coders. We should keep the `Coder` persona in mind as we go, but it is not the only Persona to consider. ⬅️ is something to consider in terms of general community engagement. But, think about others who may be interested in this work who are not Coders. (Also, why I had a hard time assimilating your comments about not leveraging a database, because ease-of-use --- if Coders are in mind --- then additional technical tools shouldn't be an issue. 

Nevertheless, for a dialogue, start an Issue (prior to writing code) - develop the dialogue there (this is a natural part of the lifecycle of a Feature Request), let the idea evolve, and hopefully at some point, the Issue can be re-cast into a clearly defined scope of work that delivers concrete user value. 

---

Regarding Focus and Scope - I set out to to clean up Issues that cannot be completed. I have shared User Stories describing the value of proposed Features, and Acceptance Criteria attempting to clarify the user experience desired - Acceptance Criteria is also pseudo-code for automated tests.

---

Sounds like a meeting will be helpful to increase clarity and focus on the backlog of Issues. I parsed out many comments in your GraphQL Issue into discrete Issues that can be tackled one at a time.

Let's find a time to chat soon.
